docs/capacity-planning.rst

```diff
@@ -0,0 +1,326 @@
+
+**KSQL Command Topics**
+
+When run in interactive configuration, a KSQL cluster creates an internal topic (whose name is derived from the ksql.service.id setting) to persist the log of queries to run across all the servers in this KSQL cluster. These special-purpose topics for KSQL are called command topics.  Command topics have a single partition and default to a replication factor of 1.
+
+Consumption/Production
+++++++++++++++++++++++
+
+You may need to provision additional Kafka brokers to accommodate KSQL production/consumption to/from your Kafka cluster.
+
+At minimum, each query consumes each record from an input Kafka topic and produces records to an output Kafka topic.
+
+Some queries require that the input Stream be repartitioned so that all messages being aggregated or joined together reside in the same partition. Each repartition produces and consumes every record.
+
+Finally, stateful queries such as aggregations and joins produce records to a changelog topic for their respective state stores. 
+
+Important Sizing Factors
+========================
+
+In this section we’ll go over some of the important factors to consider when scoping out your KSQL deployment.
+
+**Throughput**: In general, higher throughput requires more resources.
+
+**Query Types**: Your realized throughput will largely be a function of the type of queries you run. You can think of KSQL queries as falling into three categories:
+
+    1. Project/Filter, e.g. ``SELECT <columns> FROM <table/stream> WHERE <condition>``
+    2. Joins
+    3. Aggregations, e.g. ``SUM, COUNT, TOPK, TOPKDISTINCT``
+
+A project/filter query reads records from an input stream or table, may filter the records according to some predicate, and performs stateless transformations on the columns before writing out records to a sink stream or table. Project/filter queries require the fewest resources. For a single project/filter query running on an instance provisioned as recommended above you can expect to realize from ~40 MB/second up to the rate supported by your network. The throughput largely depends on the average message size and complexity. Processing small messages with many columns is CPU intensive and will saturate your CPU. Processing large messages with fewer columns requires less CPU and KSQL will start saturating the network for such workloads.
+
+Stream-table joins read and write to Kafka Streams state stores and require around twice the CPU of project/filter. Though Kafka Streams state stores are stored on disk, we recommend that you provision sufficient memory to keep the working set memory-resident to avoid expensive disk i/os. So expect around half the throughput and expect to provision higher-memory instances.
```

Comment:
avoid using "we": `Though Kafka Streams state stores are stored on disk, it is recommend that you provision...`